src/remote.c

```diff
@@ -191,3 +198,57 @@ int remote_get(const char* url, uint8_t** buf, size_t* len, int* b_int, int32_t*
   ESP_LOGI(TAG,"fetched new webp");
   return 0;
 }
+
+
+bool is_local_address(const char* url) {
+  // Skip protocol prefix if present
+  const char* http_prefix = "http://";
+  const char* https_prefix = "https://";
+
+  if (strncmp(url, http_prefix, strlen(http_prefix)) == 0) {
+    url += strlen(http_prefix);
+  } else if (strncmp(url, https_prefix, strlen(https_prefix)) == 0) {
+    url += strlen(https_prefix);
+  }
+
+  // Make a copy of the host part (without the port)
+  char host[256];
+  strncpy(host, url, sizeof(host) - 1);
+  host[sizeof(host) - 1] = '\0';
+
+  // Remove port number if present
+  char* port = strchr(host, ':');
+  if (port != NULL) {
+    *port = '\0';  // Terminate string at the colon
+  }
+
+  // Check for .local TLD
+  size_t len = strlen(host);
+  if (len >= 6 && strcmp(host + len - 6, ".local") == 0) {
+    return true;
+  }
+
+  // Check if starts with common local IP prefixes
+  if (strncmp(host, "10.", 3) == 0) {
```

Comment:
TLS certificates for local IP addresses? This doesn't make a lot of sense.
Certificates technically can be issued for IP addresses instead of FQDNs, but this is rare and it's only for public addresses.

Reply:
Huh? You want `is_local_address` to return `true` for publicly routable IP addresses? I really don't think this is worth spending time on. Disabling TLS verification is worse than just not using it because it gives some false sense of security.

Reply:
It's not disabled though. It only skips if you don't set the cert_bundle.
![Uploading Screenshot 2025-04-08 at 12.05.11 PM.png…]()

Reply:
I understand that, but that flag is still only meant for testing and should not be set by default.
There's no good way to determine whether a given hostname is "local" or not. `.local` is not a TLD (https://en.wikipedia.org/wiki/.local) and only used for mDNS in a local network, but there can be infinitely other local domains (`.lan` is common, mine is `.localdomain`) or just plain hostnames resolved against a search domain delivered via DHCP.
Even if you resolve the hostname via DNS and look at whether the resulting IP (v4 and/or v6) are local, it's vulnerable to DNS spoofing.
Because there's no good way to do that, you can't decider when to supply the default CAs and when to skip verification.

Reply:
so what's the solution for someone who wants to use https on a local address ?

Reply:
Either use a valid certificate from a default CA (e.g. using the DNS-01 challenge), provide a custom CA certificate to the firmware, or use HTTP (without the S).

I personally use the first option, for example.